src/pages/BlendPoolSelectPage.tsx

```diff
@@ -141,25 +176,82 @@ export default function BlendPoolSelectPage(props: BlendPoolSelectPageProps) {
     isMounted.current = true;
     return () => {
       isMounted.current = false;
-    }
+    };
   }, []);
 
   const loadData = useCallback(async () => {
     let poolData = Array.from(poolDataMap.values()) as BlendPoolMarkers[];
+    const poolRequests = poolData.map(async (pool) => {
+      return {
+        poolData: pool,
+        poolStats: await axios.get(
+          `${API_URL}/pool_stats/${pool.poolAddress}/1`,
+          {
+            transformResponse: (response) => {
+              const responseJSON = JSON.parse(response)[0];
+              return responseJSON as OffChainPoolStats;
+            },
+          }
+        ),
+      };
+    });
+    const poolStatsDataResponse = await Promise.all(poolRequests);
+    // Fetch the intermediate data (everything except the timestamp)
+    // TODO: see if we can move move the timestamp call into the above promise.all
+    const intermediatePoolStatsData = poolStatsDataResponse.map(
+      (response: {
+        poolData: BlendPoolMarkers;
+        poolStats: AxiosResponse<any, any>;
+      }) => {
+        return {
+          poolData: response.poolData,
+          poolStats: response.poolStats.data,
+        };
+      }
+    );
+    // Fetch the timestamps for all the pools
+    const factoryCreationEventsRequest = await makeEtherscanRequest(
+      0,
```

Comment:
We don't need to search all the way from block 0. Can start from the block at which the Blend Factory was deployed